Arduino.mk

```diff
@@ -1605,7 +1605,7 @@ ifneq ($(words $(wildcard $(TAGS_FILE))), 0)
 	rm -f $(TAGS_FILE)
 endif
 	@$(ECHO) "Generating tags for local sources (INO an PDE files as C++): "
-	$(CTAGS_CMD) $(TAGS_FILE) --langmap=c++:.ino --langmap=c++:.pde $(LOCAL_SRCS)
+	$(CTAGS_CMD) $(TAGS_FILE) --langmap=c++:+.ino.pde $(LOCAL_SRCS)
```

Comment:
is this right, i don't have any experience of ctags, but surely there should be a space between .ino and .pde and no + ? or is it just a weird syntax ;)

Reply:
It's a complicated syntax because you can comma separate languages and assiated filetypes. That is why there is no space or comma between the filetypes; the character would be used to say add java filetypes. I got it wrong in the first instance by assuming it appended, not overwrote. 

[from ctags reference page:](http://ctags.sourceforge.net/ctags.html)
>−−langmap=map[,map[...]] 
...If the first character in a map is a plus sign, then the extensions and file name patterns in that map will be appended to the current map for that language; otherwise, the map will replace the current map. For example, to specify that only files with extensions of .c and .x are to be treated as C language files, use "−−langmap=c:.c.x"; to also add files with extensions of .j as Java language files, specify "−−langmap=c:.c.x,java:+.j"....